tests/datalab/test_datalab.py

```diff
@@ -871,6 +873,39 @@ def test_incremental_search_using_pred_probs(self, lab, sorted_embeddings):
         assert non_iid_summary["score"].values[0] == 0
         assert non_iid_summary["num_issues"].values[0] == 1
 
+    def test_non_iid_issues_pred_probs_knn_graph_checks(self, lab, random_embeddings):
+        """
+        Note
+        ----
+        If the user did provides `features` or there was already a KNN graph
+        constructed in Datalab, the results should be returned as they currently
+        are, not using the `pred_probs` at all.
+        """
+        pred_probs = pred_probs_from_features(random_embeddings)
+        # knn graph is computed and stored
+        lab.find_issues(
+            features=random_embeddings, pred_probs=pred_probs, issue_types={"outlier": {}}
+        )
+        cached_knn_graph = lab.get_info("statistics").get("weighted_knn_graph")
+        assert cached_knn_graph is not None
+        lab.find_issues(pred_probs=pred_probs, issue_types={"non_iid": {}})
+        knn_graph_after_non_iid = lab.get_info("statistics").get("weighted_knn_graph")
+        # Check if stored knn graph is same as before
+        assert cached_knn_graph is knn_graph_after_non_iid
+        issues_1 = lab.get_issues("non_iid")
```

Comment:
@elisno 
Should we check if `issues_1` matches with the result of 

```python
lab.find_issues(
            features=random_embeddings, pred_probs=pred_probs, issue_types={"non_iid": {}}
        )
```
i.e. ensure that 
`results_with_features == results_with_cached_knn == results_with_passed_knn`?

Reply:
It's not strictly necessary in this test.
At this point, the condition `results_with_features == results_with_cached_knn` should be true for various issue managers.
Let's not worry about it in this PR.